src/java/org/apache/cassandra/hadoop/ColumnFamilySplit.java

```diff
@@ -17,45 +17,45 @@
  */
 package org.apache.cassandra.hadoop;
 
-import org.apache.hadoop.io.Writable;
-import org.apache.hadoop.mapreduce.InputSplit;
-
 import java.io.DataInput;
 import java.io.DataOutput;
 import java.io.IOException;
-import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.cassandra.dht.IPartitioner;
+import org.apache.cassandra.dht.Range;
+import org.apache.cassandra.dht.Token;
+import org.apache.cassandra.exceptions.ConfigurationException;
+import org.apache.cassandra.utils.FBUtilities;
+
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.InputSplit;
+
 
 public class ColumnFamilySplit extends InputSplit implements Writable, org.apache.hadoop.mapred.InputSplit
 {
-    private String startToken;
-    private String endToken;
+    private List<Range<Token>> tokenRanges;
     private long length;
     private String[] dataNodes;
+    private String partitionerClassname;
 
-    @Deprecated
-    public ColumnFamilySplit(String startToken, String endToken, String[] dataNodes)
```

Comment:
> Why do you pass partitionerClassname in the split?

i need a way to serdes the token ranges.
 is there a better way than ?

    Token.TokenFactory factory = FBUtilities.newPartitioner(partitionerClassname).getTokenFactory();
    out.writeUTF(factory.toString(range.left));

> not too sure what to do with this… presume the random partitioner is still being used?

i ask because this breaks compatibility against third-party code. 
the deprecated constructor can be removed i'm sure, but the non-deprecated constructor below changes signature.

is it ok to introduce this api breakage?
or should i add the existing constructor, deprecated, with presumption the partitioner is a RandomPartitioner ?